strawberry/schema/types/base_scalars.py

```diff
@@ -2,12 +2,25 @@
 import decimal
 import uuid
 from operator import methodcaller
+from typing import Callable
 
 import dateutil.parser
 
+from graphql import GraphQLError
+
 from strawberry.custom_scalar import scalar
 
 
+def wrap_iso_parser(parser: Callable, _type: str) -> Callable:
+    def inner(value: str):
+        try:
+            return parser(value)
+        except ValueError as e:
+            raise GraphQLError(f'Value cannot represent a {_type}: "{value}". {e}')
+
+    return inner
```

Comment:
```suggestion
def wrap_iso_parser(parser: Callable, type_: str) -> Callable:
    def inner(value: str):
        try:
            return parser(value)
        except ValueError as e:
            raise GraphQLError(f'Value cannot represent a {type_}: "{value}". {e}')

    return inner
```

Standard convention to avoid shadowing is underscore after var name.

Reply:
TIL